src/daft-local-plan/src/plan.rs

```diff
@@ -48,6 +48,8 @@ pub enum LocalPhysicalPlan {
     CatalogWrite(CatalogWrite),
     #[cfg(feature = "python")]
     LanceWrite(LanceWrite),
+    #[cfg(feature = "python")]
+    CustomWrite(CustomWrite),
```

Comment:
What if  instead we call it a `WriteSink` Custom is a bit hard to follow

Reply:
Sounds good. Opted for DataSink over WriteSink to keep naming consistent (and there's already a WriteSink defined in swordfish today)